fetch.io/index.d.ts

```diff
@@ -0,0 +1,132 @@
+// Type definitions for fetch.io 3.1
+// Project: https://github.com/haoxins/fetch.io
+// Definitions by: newraina <https://github.com/newraina>
+// Definitions: https://github.com/DefinitelyTyped/DefinitelyTyped
+
+/// <reference types="whatwg-fetch" />
+
+  type TUrl = string;
+
+  type TMethod = 'delete' | 'get' | 'head' | 'options' | 'post' | 'put';
+
+  interface Query {
+    [key: string]: number | boolean | string;
+  }
+
+  interface Header {
+    [key: string]: string;
+  }
+
+  interface Options extends RequestInit {
+
+    prefix?: string;
+
+    query?: Query;
+
+    header?: Header;
+
+    beforeRequest?(url: TUrl, body: BodyInit): boolean;
+
+    afterResponse?(res: Response): void;
+
+    afterJSON?(body: any): void;
+  }
+
+  declare namespace FetchIo {
+
+    class Request {
+
+      constructor(method: TMethod, url: TUrl, options: Options)
+
+      /**
+       * HTTP delete method
+       */
+      delete: (url: TUrl) => this;
+
+      /**
+       * HTTP get method
+       */
+      get: (url: TUrl) => this;
+
+      /**
+       * HTTP head method
+       */
+      head: (url: TUrl) => this;
+
+      /**
+       * HTTP options method
+       */
+      options: (url: TUrl) => this;
+
+      /**
+       * HTTP post method
+       */
+      post: (url: TUrl) => this;
+
+      /**
+       * HTTP put method
+       */
+      put: (url: TUrl) => this;
+
+      /**
+       * Set Options
+       */
+      config(key: string, value: any): this
+
+      config(opts: {[key: string]: any}): this
+
+      /**
+       * Set Header
+       */
+      set(key: string, value: any): this
+
+      set(opts: {[key: string]: any}): this
+
+      /**
+       * Set Content-Type
+       */
+      type(type: 'json' | 'form' | 'urlencoded'): this
+
+      /**
+       * Add query string
+       */
+      query(object: {[key: string]: any}): this
+
+      /**
+       * Send data
+       */
+      send(data: {[key: string]: any}): this
+
+      /**
+       * ppend formData
+       */
+      append(key: string, value: string): this
+
+      /**
+       * Get Response directly
+       */
+      then(resolve: (value?: Response) => void, reject?: (reason?: any) => void): Promise<any>
+
+      then<T>(resolve: (value?: Response) => void, reject?: (reason?: any) => void): Promise<T>
```

Comment:
Please remove this overload - avoid type arguments that aren't used in parameters.

From [Common mistakes](https://github.com/DefinitelyTyped/DefinitelyTyped#common-mistakes):

> `getMeAT<T>(): T`:
    If a type parameter does not appear in the types of any parameters, you don't really have a generic function, you just have a disguised type assertion.
    Prefer to use a real type assertion, e.g. `getMeAT() as number`.
    Example where a type parameter is acceptable: `function id<T>(value: T): T;`.
    Example where it is not acceptable: `function parseJson<T>(json: string): T;`.
    Exception: `new Map<string, number>()` is OK.